tests/React.Tests/Router/HtmlHelperExtensionsTest.cs

```diff
@@ -0,0 +1,212 @@
+﻿/*
+ *  Copyright (c) 2014-Present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+using Moq;
+using Xunit;
+using React.Exceptions;
+using React.Router;
+using React.Tests.Core;
+using System.Web;
+
+namespace React.Tests.Router
+{
+	public class HtmlHelperExtensionsTest
+	{
+		/// <summary>
+		/// Creates a mock <see cref="IReactEnvironment"/> and registers it with the IoC container
+		/// This is only required because <see cref="HtmlHelperExtensions"/> can not be
+		/// injected :(
+		/// </summary>
+		private ReactEnvironmentTest.Mocks ConfigureMockReactEnvironment()
+		{
+			var mocks = new ReactEnvironmentTest.Mocks();
+
+			var environment = mocks.CreateReactEnvironment();
+			AssemblyRegistration.Container.Register<IReactEnvironment>(environment);
+			return mocks;
+		}
+
+		private Mock<IReactEnvironment> ConfigureMockEnvironment()
+		{
+			var environment = new Mock<IReactEnvironment>();
+			AssemblyRegistration.Container.Register(environment.Object);
+			return environment;
+		}
+
+		private Mock<IReactSiteConfiguration> ConfigureMockConfiguration()
+		{
+			var config = new Mock<IReactSiteConfiguration>();
+			AssemblyRegistration.Container.Register(config.Object);
+			return config;
+		}
+
+		//[Fact]
+		//public void EngineIsReturnedToPoolAfterRender()
```

Comment:
> It's a static extension method calling another static extension method. Is there a way to proxy this behavior and have CreateRouterComponent create a Mocked reactcomponent?

Hmm, that's tricky 😕 Static methods can't be mocked. Are you having issues with the test because of that?